ui/src/components/input/__tests__/QInput.cy.js

```diff
@@ -1,92 +1,773 @@
+              maxlength: 10
+            }
+          })
+
+          getHostElement()
+            .get('input')
+            .type('1234567890abcdefghij')
+            .should('have.value', '1234567890')
         })
       })
     })
 
     describe('Category: style', () => {
       describe('(prop): input-class', () => {
-        it.skip(' ', () => {
-          //
+        it('should apply the input class to the underlying element', () => {
+          mountQInput()
+          getHostElement().get('input.input-class').should('not.exist')
+
+          mountQInput({
+            props: {
+              inputClass: 'input-class'
+            }
+          })
+
+          getHostElement().get('input.input-class').should('exist')
+        })
+
+        it('should apply the input class specified as an object', () => {
+          mountQInput({
+            props: {
+              inputClass: { 'input-class': true }
+            }
+          })
+
+          getHostElement().get('input.input-class').should('exist')
         })
       })
 
       describe('(prop): input-style', () => {
-        it.skip(' ', () => {
-          //
+        it('should set input-style as string', () => {
+          mountQInput({
+            props: {
+              inputStyle: 'background-color: red'
+            }
+          })
+
+          getHostElement().get('input[style="background-color: red;"]').should('exist')
+        })
+
+        it('should set input-style as an object', () => {
+          mountQInput({
+            props: {
+              inputStyle: { backgroundColor: 'red' }
+            }
+          })
+
+          getHostElement().get('input[style="background-color: red;"]').should('exist')
+        })
+      })
+
+      describe('(prop): label-color', () => {
+        it('should display a label color', () => {
+          const label = 'Hello there!'
+          mountQInput({
+            props: {
+              label,
+              labelColor: 'red'
+            }
+          })
+
+          getHostElement().get('.q-field__label.text-red').should('contain', label)
+        })
+      })
+
+      describe('(prop): color', () => {
+        it('should set a color on the component', () => {
+          mountQInput()
+          getHostElement().get('.q-field__control.text-red').should('not.exist')
+          mountQInput({
+            props: {
+              color: 'red'
+            }
+          })
+
+          getHostElement().get('input').then(() => {
+            Cypress.vueWrapper.vm.focus()
+            getHostElement().get('.q-field__control.text-red').should('exist')
+          })
+        })
+      })
+
+      describe('(prop): bg-color', () => {
+        it('should display a background color', () => {
+          mountQInput({
+            props: {
+              bgColor: 'red'
+            }
+          })
+
+          getHostElement().get('.q-field__control.bg-red').should('exist')
+        })
+      })
+
+      describe('(prop): filled', () => {
```

Comment:
Check out if it's possible to create those tests with a cycle or if we already did that in other component tests

e.g.
```ts
const fieldApparences = ['filled', 'outlined', ...];

for (const fieldStyle of fieldStyles) {
  describe(....)
}
```

We didn't into QBtn as far as I can tell, can't remember why, check out the review of the PR which inserted those